### DISCUSSION TITLE
Help on wiring for PIN mapping

### DISCUSSION THREAD

**Discussion by anktababa** [tbnobody/OpenDTU#1288] [Q&A]
Hi folks,

Thanks to the documentation i have been able to flash and configure OpenDTU on the ESP 32 WROOM.
Now i am trying to understand how to configure the PIN mapping.

I have a ESP 32 WROOM, SSD1306 display and NRF24L01Plus.
My wiring is as follow : 
https://cdn.eazyauction.de/90/908719ce153a213b87987806f5795c7e907ae2e19a4a28db71b759fb194d79e737505b3f42b379d057baa3e78c3e5d40/img/1500x1500_a974576903a418068a2790c3c4a2c11fd779a23d.jpeg 

After reading the doc i understand that i should upload a file pin_mapping.json which contains the PIN that are used.
My issue is that i don't understand the PIN name as sometime this is a pin number or RX2, TX2 etc. 
On top of that the NRF24L01Plus device doesn't have the same PIN name as the documentation so i am confused about the PIN map.
            "miso" --> MI
            "mosi"--> MO
            "clk" --> SCK ?
            "irq" --> IRQ
            "en" --> CE ?
            "cs" --> CSN ?
Here is my current PIN file (for the screen i think it is ok but for nrf24 i don't think): 
[
    {
        "name": "Generic NodeMCU 38 pin with SSD1306",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
        }
    },
]

Thank you for your help.

**Comment by wib100** [comment]
There are already predefined files in the project folder: docs/DeviceProfiles
Please check the docs/DeviceProfiles.md for more information. 
Hope this help!

**Comment by anktababa** [comment]
Thanks for your answer.
After spending some hours on it, it works !
The device profile included are helping a lot.

I still have an issue which is not understable, I wanted to do another one for my father but so far without success.
- My ESP32 WROM 32 30 PIN
- SSD1306
- NRF24L01PLUS (external antenna)

I bought the same devices to do the one for my father, the only difference is :
- i bought 3 x ESP32 WROM 32 38 PIN from AZDelivery
- 3 x SSD1306
- 3 x NRF24L01PLUS (external antenna)
The only difference is 30 PIN vs 38 PIN.

My cabling is the same on all boards (same interface PIN), it work on my 30 PIN it doesn't work on all my 38Pins. 
The config is the same, the device info is good 
nRF24 | configured
status connected
type nRF24L01+

Does the inverter only accept to talk to the same DTU ID ? I guess only one at a time but when i turn off the one that is working other should be ok ?
The 3 devices that are not working are running the last version while the one that is working is running [v23.8.22].
Thanks for the inputs.

**Reply by Tiese** [reply]
Vielleicht hilft das hier weiter...
Maybe this will help...
[PINS](https://www.studiopieters.nl/esp32-pinout/)

**Comment by anktababa** [comment]
Thanks Tiese, this is instructive i learnt good things about the ESP32. 
Unfortunately even after reading i am not able to explain my issue.

Let me try to sum up, may it will also help other people 👍 

I bought
-3 x ESP32-WROM32-38PIN from Azdelivery (as shown on the github page)
- 3 x NRF24L01 + PA+
- 3 x 0.96" SSD1306 display from Azdelivery
- Dupont lambda cable

After reading the github and the subject here, I have the impression of having wired everything as requested.

Operation carried out 3 times (because 3 units of each), the result is always the same: OpenDTU is ok, Wifi OK, Screen OK but communication to the interrupter KO. I checked the NTP, it's ok, the serial number too, my DTU is set at 2M.
In the System Info page everything is good:

![image](https://github.com/tbnobody/OpenDTU/assets/17125233/df91ae26-9039-4a59-b8e0-b5558d6f5396)

My only idea remains that I am making a big, stupid mistake with the wiring, so here is the configuration in OpenDTU:
![image](https://github.com/tbnobody/OpenDTU/assets/17125233/76f2b5ee-e567-4ee0-93e2-500e7e16a7ff)

![image](https://github.com/tbnobody/OpenDTU/assets/17125233/176903ad-5a8b-4e03-9d46-4d54a28f42dc)

![image](https://github.com/tbnobody/OpenDTU/assets/17125233/00c9c5f9-7349-482d-b9c4-93ae4dcae0f4)

The physical assembly (screen wire removed to try to make it more "clear"):
Top view of the 38 pin ESP32:
![image](https://github.com/tbnobody/OpenDTU/assets/17125233/a2cd21f9-6ec0-4e42-ac2b-51ab46f089d4)

Profile view allowing you to see the color of the wires and the PIN:
![image](https://github.com/tbnobody/OpenDTU/assets/17125233/170c111e-5bd8-4894-b999-3b9ad43da755)

View of the nrf with the wires and pins :
![image](https://github.com/tbnobody/OpenDTU/assets/17125233/1add51f5-d6f3-4bc5-9d9b-0f1e9c93b028)

Which give :
NRF24: MISO --> G19 (Orange wire)
NRF24: IRQ --> G16 (Green Wire)
NRF24: SCK --> G18 (Taupe Thread)
NRF24: MOSI --> G23 (Yellow Wire)
NRF24: CE --> G4 (Blue Wire)
NRF24: CSN --> G5 (Purple Wire)
NRF24: GND --> GND (top right) (Black Wire)
NRF24: VCC --> 3V3 (top left) (Red Wire)

    {
        "name": "NRF24 with SSD1306",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
        }
       

I hope this can help those who want to get started, if I find a clue/explanation I will share, in the meantime if anyone has an idea on the origin of my problem/error I am interested.

Thank you.

**Comment by anktababa** [comment]
Hello, 
I tried to flash again the ESP32.

I used the release v23.8.22 as it was the one that is working on my ESP32 30Pin.
After flashing this release and restoring the config of the ESP32 it works. (without touching the cabling etc)
I will try to do it on a second unit that was not working to see if i have the same result, if yes i will try to reflash the third unit using the last version and see.

**Comment by anktababa** [comment]
Hello, 
After flashing again with the last version and the previous one, it works. 
It seems that it was a flashing issue for whatever reason.
Related to the materiel the only thing that i can see is when :
- Using the YL-105 the connectivity to the inverter seems to be more stable. 
- Not using the YL-105 i can see in HA some reachable turn off, reachable turn on, during the day but when using the YL-105 i don't have that anymore.

Do you have a feedback regarding the use of YL-105 ?
While looking at the case available on thingiverse etc almost none of them is designed to use this component.
It seems that it is not used in general so i am wondering if it is only me or a general behavior.

Thanks

**Comment by lefrenchy49** [comment]
> Hi folks,
> 
> Thanks to the documentation i have been able to flash and configure OpenDTU on the ESP 32 WROOM. Now i am trying to understand how to configure the PIN mapping.
> 
> I have a ESP 32 WROOM, SSD1306 display and NRF24L01Plus. My wiring is as follow : https://cdn.eazyauction.de/90/908719ce153a213b87987806f5795c7e907ae2e19a4a28db71b759fb194d79e737505b3f42b379d057baa3e78c3e5d40/img/1500x1500_a974576903a418068a2790c3c4a2c11fd779a23d.jpeg
> 
> After reading the doc i understand that i should upload a file pin_mapping.json which contains the PIN that are used. My issue is that i don't understand the PIN name as sometime this is a pin number or RX2, TX2 etc. On top of that the NRF24L01Plus device doesn't have the same PIN name as the documentation so i am confused about the PIN map. "miso" --> MI "mosi"--> MO "clk" --> SCK ? "irq" --> IRQ "en" --> CE ? "cs" --> CSN ? Here is my current PIN file (for the screen i think it is ok but for nrf24 i don't think): [ { "name": "Generic NodeMCU 38 pin with SSD1306", "nrf24": { "miso": 19, "mosi": 23, "clk": 18, "irq": 16, "en": 4, "cs": 5 }, "eth": { "enabled": false, }, "display": { "type": 2, "data": 21, "clk": 22 } }, ]
> 
> Thank you for your help.

Hello,

I have excatly the same wiring as you have and I don't know how to set the PIN as well :(

Can you give me your file so I can check on my setup ?

Thank you very much !

**Reply by tbnobody** [reply]
Just look at the documentation: https://github.com/tbnobody/OpenDTU/wiki/NRF24L01-Module#pin-assignment
Several short names are mentioned there.

A first good start for a generic ESP32-Wroom module is the `nodemcu_esp32.json`

**Reply by lefrenchy49** [reply]
OK, I will, thanks ;)
